.github/workflows/automation-scripts/update-changelogs.js

```diff
@@ -0,0 +1,83 @@
+const fs = require('fs');
+const path = require('path');
+
+// Read all files in the directory
+function getFilePaths(directoryPath) {
+  let filePaths = [];
+  let files = fs.readdirSync(directoryPath);
+  files.forEach(file => {
+    let filePath = path.join(directoryPath, file);
+    filePaths.push(filePath);
+  });
+  return filePaths
+}
+
+function loadJsonFiles(filePath) {
+  // Load each JSON file
+  data = fs.readFileSync(filePath);
+  const jsonData = JSON.parse(data);
+  return jsonData;
+}
+
+function sortByVersion(objects) {
+  return objects.sort((a, b) => {
+    const versionA = a.version.split('.').map(Number);
+    const versionB = b.version.split('.').map(Number);
+
+    for (let i = 0; i < 3; i++) {
+      if (versionA[i] < versionB[i]) return 1;
+      if (versionA[i] > versionB[i]) return -1;
+    }
+
+    return 0;
+  });
+}
+
+function findNewEntries(currentEntries, incomingEntries) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
https://javascript.info/map-set#set

Reply:
using a set seems to not work, don't know why. It doesn't seem to identify the entries as duplicates, and treats them all as unique. I'll keep looking into it

Reply:
done